### PR TITLE
controlplane/apiserver: Trim managedFields off self-requested informers

### PR DESCRIPTION
**Pull request by linxiulei**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

Improve memory efficiency for `kube-apiserver`

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:

Similar work was done in https://github.com/kubernetes/kubernetes/pull/118455 for `kube-controller-manager`

#### Does this PR introduce a user-facing change?

```release-note
Improve memory usage of kube-apiserver by dropping the `.metadata.managedFields` field that self-requested informers of kube-apiserver doesn't need.
```

### CODE REVIEW

**Comment by linxiulei** [comment]
ping @wojtek-t

File: pkg/controlplane/apiserver/config.go [hunk 1]
```diff
@@ -135,7 +136,13 @@ func BuildGenericConfig(
 		lastErr = fmt.Errorf("failed to create real external clientset: %w", err)
 		return
 	}
-	versionedInformers = clientgoinformers.NewSharedInformerFactory(clientgoExternalClient, 10*time.Minute)
+	trim := func(obj interface{}) (interface{}, error) {
+		if accessor, err := meta.Accessor(obj); err == nil && accessor.GetManagedFields() != nil {
```
[the remark this answers is not in the document]
Comment (linxiulei) [reply pkg/controlplane/apiserver/config.go:140]: That's right. https://github.com/kubernetes/kubernetes/issues/119658 was the check to warn the consumption.

[the remark this answers is not in the document]
Comment (linxiulei) [reply pkg/controlplane/apiserver/config.go:140]: I don't think it'll be really hard if we have a good reason to add it back. This change will reduce 15% memory overhead for informer cache, but it should be around 5% reduction of total apiserver memory overhead for common cases. So 5% regression might not be a hard decision. However, the 5% is extra gain until we change our minds.

I believe, in fact, as like many software lifecycles, k8s has been organically adding features and whatnot that may bring in minor performance regression.

[the remark this answers is not in the document]
Comment (jpbetz) [reply pkg/controlplane/apiserver/config.go:140]: Thanks for the numbers!  I'm OK with this in that case.

**Pull request merged**